Storage/tests/Unit/Connection/RestTest.php

```diff
@@ -540,6 +545,108 @@ public function retryFunctionReturnValues()
         ];
     }
 
+    public function retryStrategyCases()
+    {
+        return [
+            // Idempotent operation with retriable error code
+            // and never retry strategy => no retry
+            [
+                'buckets',
+                'get',
+                ['retryStrategy' => RetryTrait::$RETRY_STRATEGY_NEVER],
+                503,
+                1,
+                false
+            ],
+            [
+                'serviceaccount',
+                'get',
+                ['retryStrategy' => RetryTrait::$RETRY_STRATEGY_NEVER],
+                504,
+                1,
+                false
+            ],
+            // Idempotent operation with non retriable error code
+            // and always retry strategy => retry
+            [
+                'buckets',
+                'get',
+                ['retryStrategy' => RetryTrait::$RETRY_STRATEGY_ALWAYS],
+                400,
+                1,
+                true
```

Comment:
This case should return false, as mentioned above.

Reply:
Fixed it in [a47e376](https://github.com/googleapis/google-cloud-php/pull/5731/commits/a47e3761927e63746dafff9048bc0c8b17b74bb0)